lib/fieldhand/paginator.rb

```diff
@@ -81,10 +80,14 @@ def request(query = {})
       request_uri.query = encode_query(query)
 
       logger.info('Fieldhand') { "GET #{request_uri}" }
-      http.get(request_uri.request_uri).body
+      res = http.get(request_uri.request_uri)
+      raise NetworkError.new("Invalid response: #{res.code} #{res.msg}", res) unless res.is_a?(::Net::HTTPSuccess)
+
+      request.body
```

Comment:
Would it be clearer to prefer the unabbreviated `response` over `res` here as I believe we should be returning the `response` body, not the `request` body:

```ruby
response = http.get(request_uri.request_uri)
raise NetworkError.new("Unsuccessful response: #{response.code} #{response.msg}", res) unless res.is_a?(::Net::HTTPSuccess)

response.body
```